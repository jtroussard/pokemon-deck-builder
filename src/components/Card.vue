<template>
        <span>
            <div v-if="card" class='card'>
                <p class='card-name'>{{ card.name.toUpperCase() }}</p>
                <img :src="`${card.sprites.front_default}`">
                <p class="card-stat">weight: {{ card.weight }}</p>
            </div>
        </span>
</template>

<script>
    export default {
        name: 'Card',
        props: ['card'],
        methods: {
            sum() {
                return 1
            },
            summing() {
                return 100
            },
            newmethod() {
                return null
            }
        },
    }
</script>

<style scoped>

img {
    width: 100%;
    height: auto;
}

.card {
    display: inline-block;
    width: 150px;
    height: 210px;
    border: solid 2px rgb(87, 87, 87);
    border-radius: 4pt;
    padding-top: 10px;
    margin-right: 5%;
    position: relative;
    text-align: center;
    vertical-align: middle;
}

.card-name {
    margin-left: 5px;
    margin-right: 5px;
}

.card-stat {
    position:absolute;
    bottom:0;
    width: 150px;
    margin-bottom: 2px;
}

</style>