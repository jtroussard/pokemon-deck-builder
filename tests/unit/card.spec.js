import { mount } from '@vue/test-utils'
import Card from '../../src/components/Card'

test('Card component snapshot test', () => {
    const wrapper = mount(Card, {
        propsData: {
            card: {
                name: 'Bobo',
                weight: 100,
                sprites: {
                    front_default: 'https://bobothepokemon.com',
                },
            },
        }
    })
    expect(wrapper).toMatchSnapshot()
})