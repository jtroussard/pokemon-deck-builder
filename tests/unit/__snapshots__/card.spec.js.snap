// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component snapshot test 1`] = `<span><div class="card"><p>Bobo</p> <p class="card-name">BOBO</p> <img src="https://bobothepokemon.com"> <p class="card-stat">weight: 100</p></div></span>`;
